Source/WTF/wtf/Expected.h

```diff
@@ -186,9 +36,7 @@ inline namespace fundamentals_v3 {
 #include <wtf/StdLibExtras.h>
 #include <wtf/Unexpected.h>
 
-namespace std {
-namespace experimental {
-inline namespace fundamentals_v3 {
+namespace WTF::clang_std_copy {
```

Comment:
Note to reviewers: Extending the `namespace std` is generally UB (https://en.cppreference.com/w/cpp/language/extending_std), so I'm trying to move away from it.
Naming suggestions welcome.

Reply:
How about replacing `std` with `WTF` but keeping the sub namespaces. The api is exposed to the global namespace with a `template<class E> using Unexpected = std::experimental::unexpected<E>;` (or equivalent) anyways.

Reply:
Putting things in namespace std is indeed a bad idea.  Why don't we just put it in namespace WTF?

Also, why don't we switch from c++2a to c++2b and just have WTF::Expected be a using statement on platforms that have it?  (that would be a separate PR or two or three)

Reply:
"why don't we switch from c++2a to c++2b" -- makes sense, but it feels more like a whole-project config change I'm too scared to touch!
So unless you/someone volunteers to do it soon, I'd like to go ahead with modernizing un/expected.
Worst case, my changes here will be gone, but at least in the meantime we'll have change its uses already to match the C++23 API so it will be easier to switch to `<expected>` proper.

Reply:
> Why don't we just put it in namespace WTF?

It feels to me like it should stay in its own world, as it's a non-permanent implementation detail of our `Unexpected`, and we wouldn't want people to start referencing it directly by mistake.

> How about replacing std with WTF but keeping the sub namespaces"

"experimental" and "fundamentals_v3" are not correct anymore for this updated code, the C++23 spec is final.

Reply:
I've gone for `WTF::detail_from_clang_libstdcpp` to both advertise the detail-ness of the code (so it's shouldn't be used directly) and its provenance.